### PR TITLE
Avoid mutating the inputs

### PR DESCRIPTION
**Pull request by rharriso**
## Observed Bug

The visitor was mutating nested props. Specifically, a `key` property was
being added to on Multitenant app. If the state was made immutable, this
would throw an exception.

## Solution

Copy the results of the converter, add the `key` to that copy.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "prepublish": "npm run build",
     "prepush": "npm test",
     "pretest": "npm run lint",
-    "test": "NODE_ENV=test nyc mocha"
+    "test": "NODE_ENV=test nyc mocha",
+    "tdd": "NODE_ENV=test nyc mocha --watch"
```
Comment (pgoldrbx) [inline package.json]: trivial, i also like to add `--break` in TDD as well to focus on the first error

File: src/JsonmlToReact.js [hunk 2]
```diff
@@ -36,26 +36,19 @@ export default class JsonmlToReact {
...
-
-    // reassign key in case `converter` removed it
-    props.key = props.key || index;
+    const resultProps = result.props || rawAttrs || {};
+    const props = Object.assign({}, resultProps, {
+      className: resultProps.className || resultProps.class,
+      class: undefined,
+      key: index,
+      style: rawAttrs.style && toStyleObject(rawAttrs.style)
+    });
```
Comment (pgoldrbx) [inline src/JsonmlToReact.js]: v.minor - this potentially adds an enumerable `style: undefined` to the object. i'm not sure if we care though.

File: test/src/JsonmlToReact.spec.js [hunk 3]
```diff
@@ -158,7 +158,12 @@ describe('JsonmlToReact class', function () {
 
       jsonmlToReact.convert(node);
 
-      expect(spy.calledWith('p', { key: 0 }, 'i am a text node')).to.be.true;
+      expect(spy.calledWith('p', {
+        className: undefined,
+        class: undefined,
+        key: 0,
+        style: undefined
+      }, 'i am a text node')).to.be.true;
```
Comment (pgoldrbx) [inline test/src/JsonmlToReact.spec.js]: trivial - might be nice to assign the props object to a variable

Comment (pgoldrbx) [reply test/src/JsonmlToReact.spec.js]: minor - I think we should add a test that verifies that none of these new props are sent to a `converter`?

File: package.json (hunk 1 above)
Comment (rharriso) [reply package.json]: updated
